std/format.d

```diff
@@ -8427,42 +8430,1050 @@ printFloat_done:
+        // because this makes it much more easy to implement the division by 10.
+        int count = exp / 60 + 1;
+
+        // saved in big endian format
+        ulong[] mybig = bigbuf[0 .. count];
+
+        // only the first or the first two ulongs contain the mantiassa. The rest are zeros.
+        int lower = 60 - (exp - T.mant_dig + 1) % 60;
+        if (lower < T.mant_dig)
+        {
+            mybig[0] = mnt >> lower;
+            mybig[1] = (mnt & ((1L << lower) - 1)) << 60 - lower;
+        }
+        else
+            mybig[0] = (mnt & ((1L << lower) - 1)) << 60 - lower;
+
+        // Generation of digits by consecutive division with reminder by 10.
+        int msu = 0; // Most significant ulong; when it get's zero, we can ignore it furtheron
+        while (msu < count - 1 || mybig[$-1] != 0)
+        {
+            ulong mod = 0;
+            foreach (i;msu .. count)
+            {
+                mybig[i] |= mod << 60;
+                mod = mybig[i] % 10;
+                mybig[i] /= 10;
+            }
+            if (mybig[msu] == 0)
+                ++msu;
+
+            buffer[--left] = cast(byte) ('0'+mod);
+        }
+
+        if (f.precision>0 || f.flHash) buffer[right++] = '.';
+
+        next = roundType.ZERO;
+    }
+    else if (exp + 61 < T.mant_dig)
+    {
+        // small number without integer digits
+        //
+        // Again this number does not fit in a ulong and we use an array of ulongs. And again we
+        // only use 60 bits, because this simplifies the multiplication by 10.
+        int count = (T.mant_dig - exp - 2) / 60 + 1;
+
+        // saved in little endian format
+        ulong[] mybig = bigbuf[0 .. count];
+
+        // only the last or the last two ulongs contain the mantiassa. Because of little endian
+        // format these are the ulongs at index 0 and 1. The rest are zeros.
+        int upper = 60 - (-exp - 1) % 60;
+        if (upper < T.mant_dig)
+        {
+            mybig[0] = (mnt & ((1L << (T.mant_dig - upper)) - 1)) << 60 - (T.mant_dig - upper);
+            mybig[1] = mnt >> (T.mant_dig - upper);
+        }
+        else
+            mybig[0] = mnt << (upper - T.mant_dig);
+
+        buffer[--left] = '0'; // 0 left of the dot
+
+        if (f.precision>0 || f.flHash) buffer[right++] = '.';
+
+        // Generation of digits by consecutive multiplication by 10.
+        int lsu = 0; // Least significant ulong; when it get's zero, we can ignore it furtheron
+        while ((lsu < count - 1 || mybig[$ - 1]!=0) && right-start - 1 < f.precision)
+        {
+            ulong over = 0;
+            foreach (i;lsu .. count)
+            {
+                mybig[i] = mybig[i] * 10 + over;
+                over = mybig[i] >> 60;
+                mybig[i] &= (1L << 60) - 1;
+            }
+            if (mybig[lsu] == 0)
+                ++lsu;
+
+            buffer[right++] = cast(byte) ('0'+over);
```

Comment:
This PR has a number of inconsistent (missing) whitespace between binary operands, but otherwise looks fine.